0.wave1_data/3.process_features/scripts/extract_sc_interstellar.py

```diff
@@ -0,0 +1,87 @@
+#!/usr/bin/env python
+# coding: utf-8
+
+# # Process single cell morphology features for CellProfiler readouts
+
+# ## Import Libraries
+
+# In[1]:
+
+
+import pathlib
+import pandas as pd
+
+import sys
+sys.path.append("../../utils/")
+import sc_extraction_utils as sc_util
+
+from pycytominer.cyto_utils import cells, output
+
+
+# ## Set up paths to CellProfiler directory and outputs
+
+# In[2]:
+
+
+# Set file and directory constants
+cp_dir = "../2.cellprofiler_analysis"
+output_dir = "data"
+platemap_file = pathlib.Path("../../metadata/Interstellar_platemap.csv")
+
+# dictionary with variables for each dilation
+dilation_info_dictionary = {
+    "dilation_25": {
+        "sql_file": "dilate25.sqlite",
+        "sc_output_file" : pathlib.Path(f"{output_dir}/dilate25_sc.csv.gz")
+    },
+    "dilation_50": {
+        "sql_file": "dilate50.sqlite",
+        "sc_output_file" : pathlib.Path(f"{output_dir}/dilate50_sc.csv.gz")
+    },
+    "dilation_75": {
+        "sql_file": "dilate75.sqlite",
+        "sc_output_file" : pathlib.Path(f"{output_dir}/dilate75_sc.csv.gz")
+    },
+}
+
+
+# In[3]:
+
+
+# Define custom linking columns between compartments
+linking_cols = {
+    "Per_TranslocatedNuclei": {
+        "Per_DilatedNuclei": "TranslocatedNuclei_Parent_DilatedNuclei",
+        "Per_Nuclei": "TranslocatedNuclei_Parent_Nuclei",
+    },
+    "Per_DilatedNuclei": {"Per_TranslocatedNuclei": "DilatedNuclei_Number_Object_Number"},
+    "Per_Nuclei": {"Per_TranslocatedNuclei": "Nuclei_Number_Object_Number"},
+}
+
+
+# In[4]:
+
+
+# Load platemap file
+platemap_df = pd.read_csv(platemap_file)
+platemap_df.head()
+
+
+# In[5]:
+
+
+for dilation, info in dilation_info_dictionary.items():
+    # set the parameters for the function as variables based on the dilation dictionary info
+    method_name = dilation
```

Comment:
Instead of creating another variable, consider referencing `dilation` directly where it is needed or renaming it to `method_name` within the loop definition. The same might apply for the two additional variables below (reference them directly where needed instead of creating a variable which is used only once).

Reply:
I will keep this for now just for comprehension for myself but it would be something to improve later.